pkg/controllers/machine_controller.go

```diff
@@ -154,12 +169,54 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 		}
 	}()
 
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		sizeLogger := r.log.WithName("size")
+		defer wg.Done()
+
+		wait.UntilWithContext(ctx, func(ctx context.Context) {
```

Comment:
don't use same name for context in anonymous function. because is extremely hard know on first look which context you use, because you can use closure principle too.